Library/Homebrew/extend/ENV.rb

```diff
@@ -3,6 +3,12 @@
 require "extend/ENV/std"
 require "extend/ENV/super"
 
+class Array
+  def to_path_s
+    map(&:to_s).uniq.select { |s| File.directory?(s) }.join(File::PATH_SEPARATOR).chuzzle
```

Comment:
It wasn't clear to me what this method did until I looked at several usages of it and worked it out — the name a bit like it creates a single file path string using a list of directories, or something. Could we consider naming it `#to_PATH`, maybe? I think that would be clearer since this kind of "PATH" is conventionally uppercase.

I'm also not sure that this belongs in ENV.rb? It's not particularly related to the ENV hash. Maybe a new extend/Array.rb file.

Actually, to be honest, I'm not sure this method belongs on Array at all — converting to a PATH string isn't super related to a generic collection of elements. It feels like there's a separate responsibility here, which suggests to me it's time for a new object.

What would you think of something like this?

```ruby
class PATH # or title case, whatever. BinPath?
  attr_reader :entries

  def initialize(entries)
    @entries = entries.to_ary
  end

  def to_str
    @entries.join(File::PATH_SEPERATOR)
  end
end
```

(You'd want to expand to add stuff like checking `directory?` and calling `chuzzle`, etc., of course.)